src/controllers/controllermanager.cpp

```diff
@@ -205,6 +217,10 @@ QList<Controller*> ControllerManager::getControllerList(bool bOutputDevices, boo
     return filteredDeviceList;
 }
 
+KeyboardController* ControllerManager::getKeyboardController() {
```

Comment:
Hi, thanks! It's updated, now using `QSharedPointer` as you proposed. However, we might also consider using it for the other controllers, since we hand out raw pointers to all of those controllers [here](https://github.com/mixxxdj/mixxx/blob/master/src/controllers/controllermanager.h#L33) and [here](https://github.com/mixxxdj/mixxx/blob/master/src/controllers/controllermanager.h#L34) as well.

The downside, though, is that we would have to retrieve the naked pointer each time when we want to connect the controller to some signal / slot with `QSharedPointer::data()`.
```C++
typedef QSharedPointer<Controller> ControllerPointer;
ControllerPointer someController(new SomeController());

connect(someController.data(), SIGNAL(signal()), 
        controllerManager, SLOT(slot()));
```
Although it doesn't look that sexy, we could probably live with it. Also because most Controller connections are made from within the Controller like `this, SLOT(slot())` or `this, SIGNAL(signal())` so those connections can stay untouched.

What do you think, shall we move everything to `QSharedPointer` (and make typedefs for each Controller type and make enumerators instantiate those instead of raw `new`s ) or let it as it is?